lebedigital/openbis/README.md

```diff
@@ -0,0 +1,140 @@
+    project: str = 'exp_project', # project in which the experiment should be saved
+    collection: str = 'exp_space/exp_project/exp_collection', # collection in which the experiment should be saved
+    parents: list[str] = [exp_parents], # parents of the experiment
+    identifier: str = '', # identifier of the sample (SPACE/PROJECT/SAMPLE_CODE)
+    permId: str = '', # permID of the sample
+    sample_object=None, # Pybis object of the sample
+    datasets: list = None, # dataset objects saved in the sample
+    dataset_codes: list = None, # codes of the dataset objects saved in the sample
+)
+```
+
+> Note: You do not have to define all of those attributes, only the name, type, metadata and the location of the Sample in the Database (space, project, collection) are necessary
+
+### An example workflow which would upload your experimental step describing your concrete experiment could be 
+
+#### 1. Define the ExpStep as you see fit
+```python. 
+from lebedigital.openbis.expstep import ExpStep
+
+concrete_experiment = ExpStep()
+
+concrete_experiment.name = 'Concrete_Experiment_1'
+concrete_experiment.type = 'EXPERIMENTAL_STEP_CONCRETE_TEST'
+concrete_experiment.metadata = {'name': 'Concrete_Experiment_1', 'date': '17:10:2022'}
+
+# define the location in the datastore
+
+concrete_experiment.space = 'SPACE'
+concrete_experiment.project = 'PROJECT'
+concrete_experiment.collection = 'SPACE/PROJECT/COLLECTION'
+```
+
+#### 2. Upload the ExpStep to the Database
+```python
+concrete_experiment.upload_expstep(o) # o = connected Interbis instance
+```
+#### Good to know:
+* To upload a sample of a given type that type has to already exist in the datastore.
+* The name of the sample has to be the same as the `$name` parameter in the metadata dictionary in order to be able to find the sample efficiently again.
+* You can also load an existing sample into an ExpStep object using the `load_sample()` method of ExpStep.
+* Every time you want to upload some Sample to the Database a Type Checker will run in order to make sure your ExpStep has been defined properly.
+* You can look up which metadata the Experimental Step Type will accept by either getting the Excel import template running the Interbis `get_metadata_import_template()` method or get the properties of an Experimental Step Type by running the Interbis `get_sample_type_properties` method.
+
+### You can now upload your datasets containing your raw or processed data files.
+
+```python
+concrete_experiment.upload_dataset(
+    o, # A connected Interbis object
+    props={
+        '$name': 'dataset_name',
+        'files': ['path_to_file'],
+        'data_type': 'PROCESSED_DATA' # or RAW_DATA. PREVIEW, ... you can see them in the dataset types
+    }
+)
+```
+
+### If you have done everything correctly you should be able to see your Experimental Step and your Datasets in the Web-View of the Database in the location you defined in the ExpStep object. 
```

Comment:
add: new object/sample of type Experimental Step XY
delete "in the ExpStep object"

Reply:
Changed